src/core/components/ChatInput.tsx

```diff
@@ -102,6 +103,16 @@ const ChatInput = ({
     }
   }, []);
 
+  const allConversations: Record<string, ConversationHistory> = JSON.parse(
+    localStorage.getItem('conversations') ?? '{}',
+  );
+
+  const currentConversation = allConversations[conversationID];
```

Comment:
I've seen a few repetitions of this in various places - might make sense to simplify and expose on app context?